### ISSUE TITLE
neovim: test-harness code should use the root init.lua

### ISSUE THREAD

**Issue by fidgetingbits** [cursorless-dev/cursorless#2454]
Now that pokey added a debug/test `init.lua` at the root folder, I think it probably makes sense that the test harness just uses that rather than maintain its own at `packages/test-harness/src/config/init.lua`? The only thing I'm not sure is if a sub package like test-harness can actually files from the parent folder or if its confined to its own workspace. I don't plan to look at this for now, so just filing so we don't forget.

Caveat is test setup will use lazy and other plugins that aren't critical too running the test tar ness.

**Comment by saidelike** [comment]
I believe @pokey added support for OSX for that already in https://github.com/cursorless-dev/cursorless/pull/2256/commits/94d2f169bc6c72bf72cac08fd4ebc238d2be37c1 by using `nvim -u /path/to/init.lua`.

I've tried doing the same with my Powershell command on Windows from vscode but without much success atm. It is weird I can't do it from vscode because I am able to load the workspace `init.lua` manually from a standalone cmd.exe (outside of vscode):

![image](https://github.com/user-attachments/assets/e427b06f-164d-4d05-95d4-d739026718e1)

I have confirmed it is the right one loaded by adding a `print("Cursorless loaded")` at the end of the workspace `init.lua` which we see above was shown.

However when I try the same in `tasks.json` I get various errors:

1. this works fine without any argument passed to `nvim` (but it loads the system one instead of the workspace one)

```json
    {
      "label": "Neovim: Launch neovim",
      "type": "process",
      "windows": {
        "command": "powershell",
        "args": [
          // TODO: Add -u here to use the custom config
          "(New-Object -ComObject WScript.Shell).Run(\"\"\"nvim\"\"\", 1, $false)"
        ]
      },
```

2. using a path relative to the workspace does not work

```
        "args": [
          // TODO: Add -u here to use the custom config
          "(New-Object -ComObject WScript.Shell).Run(\"\"\"nvim -u ${workspaceFolder}\\init.lua \"\"\", 1, $false)"
        ]
```

result:

```
 *  Executing task in folder cursorless_fork: C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe (New-Object -ComObject WScript.Shell).Run("""nvim -u C:\path\to\cursorless_fork\init.lua""", 1, $false) 

The system cannot find the file specified. (Exception from HRESULT: 0x80070002)
At line:1 char:1
+ (New-Object -ComObject WScript.Shell).Run("""nvim -u C:\work\tools\vo ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [], FileNotFoundException
    + FullyQualifiedErrorId : System.IO.FileNotFoundException
```

Same with single quotes:

```
        "args": [
          // TODO: Add -u here to use the custom config
          "(New-Object -ComObject WScript.Shell).Run(\"\"\"nvim -u '${workspaceFolder}\\init.lua' \"\"\", 1, $false)"
        ]
```

result:

 *  Executing task in folder cursorless_fork: C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe (New-Object -ComObject WScript.Shell).Run("""nvim -u 'C:\path\to\cursorless_fork\init.lua' """, 1, $false) 

The system cannot find the file specified. (Exception from HRESULT: 0x80070002)
At line:1 char:1
+ (New-Object -ComObject WScript.Shell).Run("""nvim -u 'C:\ ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [], FileNotFoundException
    + FullyQualifiedErrorId : System.IO.FileNotFoundException
```

same with double quotes:

```
        "args": [
          // TODO: Add -u here to use the custom config
          "(New-Object -ComObject WScript.Shell).Run(\"\"\"nvim -u \"${workspaceFolder}\\init.lua\" \"\"\", 1, $false)"
        ]
```

result:

```
Executing task in folder cursorless_fork: C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe (New-Object -ComObject WScript.Shell).Run("""nvim -u "C:\path\to\cursorless_fork\init.lua" """, 1, $false) 

At line:1 char:55
+ (New-Object -ComObject WScript.Shell).Run("""nvim -u "C:\work\tools\v ...
+                                                       ~
Missing ')' in method call.
At line:1 char:55
+ ... """nvim -u "C:\path\to\cursorless_fork\init.lua" """,  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Unexpected token 'C:\path\to\cursorless_fork\init.lua" """' in expression or statement.
At line:1 char:110
+ ...  -u "C:\path\to\cursorless_fork\init.lua" """, 1, $fal ...
+                                                                 ~
Missing argument in parameter list.
At line:1 char:121
+ ...  "C:\path\to\cursorless_fork\init.lua" """, 1, $false)
+                                                                         ~
Unexpected token ')' in expression or statement.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : MissingEndParenthesisInMethodCall
 ```

4. hardcoding the path does not work

```
        "args": [
          // TODO: Add -u here to use the custom config
          "(New-Object -ComObject WScript.Shell).Run(\"\"\"nvim -u C:\\path\\to\\cursorless_fork\\init.lua \"\"\", 1, $false)"
        ]
```
gives:

```
 *  Executing task in folder cursorless_fork: C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe (New-Object -ComObject WScript.Shell).Run("""nvim -u C:\path\to\cursorless_fork\init.lua """, 1, $false) 

The system cannot find the file specified. (Exception from HRESULT: 0x80070002)
At line:1 char:1
+ (New-Object -ComObject WScript.Shell).Run("""nvim -u C:\...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [], FileNotFoundException
    + FullyQualifiedErrorId : System.IO.FileNotFoundException
```

**Comment by saidelike** [comment]
Yes, test-harness can access from a parent folder. We can also move the root `init.lua` elsewhere if that makes more sense. It might also be worth saying that the `lazy` cloned repos might not be ideal so we could think about removing them, see https://github.com/cursorless-dev/cursorless/issues/2521
